docs/source/sci_api_tutorial/assign_mutant.rst

```diff
@@ -0,0 +1,193 @@
+            | There are no existing software besides EnzyHTP addressing this challenge. 
+            | A language that helps user to assign mutations is defined above.
+
+``chain_sync_list``: 
+    A list like ``[(A,C),(B,D)]`` to indicate homo-chains in enzyme ploymer (like dimer). Mutations will be **copied** to the correpondinhomo-chains as it is maybe experimentally impossible to only do mutations on one chain of a homo-dimer enzyme.
+
+``random_state``: 
+    The ``int()`` seed for the random number generator. Default value is 100.
+
+``chain_index_mapper``: 
+    need to be clarified in cases that residue index in each chain is not aligned.
+
+    | e.g.: for a pair of homo-dimer below:
+    | "A": ABCDEFG (start from 7)
+    | "B": BCDEFGH (start from 14)
+    | the chain_sync_mapper should be ``{"A":0, "B":6}`` and index conversion is done by A_res_idx - 0 + 6 = B_res_idx
+
+``if_check``
+    if or not checking if each mutation is valid. (This could be pretty slow if the mutant is >10^7 level)
+
+Example Code
+==============================================
+
+Assign mutants for a target protein
+---------------------------------------------------------
+
+In this example, we perform assign mutations on a monomer or multimer protein structure. In these structures, ``test_A.pdb``, ``test_A_B.pdb``, and ``test_A_B_C_D.pdb``, A and B are homologous chains, and C and D are homologous chains distinct from A and B.
+
+The ``stru`` is obtained by reading from a PDB file.
+
+.. code:: python
+
+    from enzy_htp.structure import PDBParser
+    import enzy_htp.mutation.api as mapi
+    #mutate a sigle chain protein
+    test_A = "test_A.pdb"
+    test_A_stru = PDBParser.get_structure(test_A)
+    test_mutation_pattern_A = (
+            "GA11A, {NA176W, PA51A},"
+            " {L56A, r:2[resi 254 around 3:all not self]*5}"
+            )
+    mutants_A = mapi.assign_mutant(test_A_stru, test_mutation_pattern_A)
+    print(mutants_A)
+
+    ##mutate a two-chain protein
+    test_A_B = "test_A_B.pdb"
+    test_A_B_stru = PDBParser.get_structure(test_A_B)
+    test_mutation_pattern_A_B = ("{GA11A, NB176W, PB51A}")
+    mutation_pattern_A_B = mapi.assign_mutant(test_A_B_stru, test_mutation_pattern_A_B, chain_sync_list=[("A", "B")], chain_index_mapper{"A": 0, "B": 0})
+    print(mutation_pattern_A_B)
+
+    #mutate a four-chain protein
+    test_A_B_C_D = "test_A_B_C_D.pdb"
+    test_A_B_C_D_stru = PDBParser.get_structure(test_A_B_C_D)
+    test_mutation_pattern_A_B_C_D = ("{TA391A, RC58A}")
+    mutation_pattern_A_B_C_D = mapi.assign_mutant(test_A_B_C_D_stru, test_mutation_pattern_A_B_C_D, chain_sync_list=[("A", "B"), ("C","D")], chain_index_mapper={"A": 0, "B": 0, "C": 0, "D": 0})
+    print(mutation_pattern_A_B_C_D)
```

Comment:
Please seperate them into 3 examples. And also explain how input is obtained in each example. (reference QZ's branch)